components/net/fetch/methods.rs

```diff
@@ -512,9 +534,24 @@ fn scheme_fetch(request: &mut Request,
                                         }
                                         let length = {
                                             let mut buffer = reader.fill_buf().unwrap().to_vec();
-                                            let buffer_len = buffer.len();
+                                            let mut buffer_len = buffer.len();
                                             if let ResponseBody::Receiving(ref mut body) = *res_body.lock().unwrap() {
-                                                body.extend_from_slice(&buffer);
+                                                let offset = usize::min({
+                                                    if let Some(end) = end {
+                                                        let remaining_bytes = end as usize - body.len();
```

Comment:
Don't we need to include `start` in this calculation? Otherwise `end` could be much larger than `body.len()` even if the range is small, and we could end up with a lot of unnecessary data in the buffer.

Reply:
Hmm... I believe we don't need to include `start` here. `remaining_bytes` is used to identify the last chunk of data and its size, which will likely be different to `FILE_CHUNK_SIZE`.

But your comment made me realize that `end` could be much larger than the file size, and so we need to narrow it down in that case.

Reply:
Imagine FILE_CHUNK_SIZE = 10, the total file is 1000 bytes long, and the range is [500, 599]. There will be 9 iterations of reading FILE_CHUNK_SIZE bytes and appending them to body. On the 10th iteration, `remaining_bytes` will be 509 (599-90), which is greater than FILE_CHUNK_SIZE.  We will then append FILE_CHUNK_SIZE bytes to body (10), which exceeds the actual range we want, and then continue appending more data until we either run out of file content, or the `end - body.len()` calculation is less than FILE_CHUNK_SIZE.

Reply:
Of course, you are absolutely right. Thanks for the detailed explanation.